Functions/TestResults.ps1

```diff
@@ -222,6 +238,131 @@ function Write-NUnitTestSuiteElements($Node, [System.Xml.XmlWriter] $XmlWriter,
+    $XmlWriter.WriteAttributeString('time', ($PesterState.Time.TotalSeconds.ToString('0.000', [System.Globalization.CultureInfo]::InvariantCulture)))
+}
+
+function Write-JUnitTestSuiteElements($Node, [System.Xml.XmlWriter] $XmlWriter, [string] $Package, [uint16] $Id) {
+    $XmlWriter.WriteStartElement('testsuite')
+
+    Write-JUnitTestSuiteAttributes -Action $Node -XmlWriter $XmlWriter -Package $Package -Id $Id
+
+    if ($Node.Actions.Type -eq 'TestCase') {
+        foreach ($t in $Node.Actions) {
+            Write-JUnitTestCaseElements -Action $t -XmlWriter $XmlWriter
+        }
+    }
+    else {
+        # testgroup
+        foreach ($t in $Node.Actions.Actions) {
+            Write-JUnitTestCaseElements -Action $t -XmlWriter $XmlWriter
+        }
+    }
+
+    $XmlWriter.WriteEndElement()
+}
+
+function Write-JUnitTestSuiteAttributes($Action, [System.Xml.XmlWriter] $XmlWriter, [string] $Package, [uint16] $Id) {
+    $environment = Get-RunTimeEnvironment
+
+    $XmlWriter.WriteAttributeString('name', $Action.Name)
+    $XmlWriter.WriteAttributeString('tests', $Action.TotalCount)
+    $XmlWriter.WriteAttributeString('errors', '0')
+    $XmlWriter.WriteAttributeString('failures', $Action.FailedCount)
+    $XmlWriter.WriteAttributeString('hostname', $environment.'machine-name')
+    $XmlWriter.WriteAttributeString('id', $Id)
+    $XmlWriter.WriteAttributeString('skipped', $Action.SkippedCount)
+    $XmlWriter.WriteAttributeString('disabled', $Action.InconclusiveCount + $Action.PendingCount)
+    $XmlWriter.WriteAttributeString('package', $Package)
+    $XmlWriter.WriteAttributeString('time', $Action.Time.TotalSeconds.ToString('0.000', [System.Globalization.CultureInfo]::InvariantCulture))
+
+    $XmlWriter.WriteStartElement('properties')
+
+    foreach ($keyValuePair in $environment.GetEnumerator()) {
+        if ($keyValuePair.Name -eq 'nunit-version') {
+            continue
+        }
+
+        $XmlWriter.WriteStartElement('property')
+        $XmlWriter.WriteAttributeString('name', $keyValuePair.Name)
+        $XmlWriter.WriteAttributeString('value', $keyValuePair.Value)
+        $XmlWriter.WriteEndElement()
+    }
+
+    $XmlWriter.WriteEndElement()
+}
+
+function Write-JUnitTestCaseElements($Action, [System.Xml.XmlWriter] $XmlWriter) {
+    $XmlWriter.WriteStartElement('testcase')
+
+    Write-JUnitTestCaseAttributes -Action $Action -XmlWriter $XmlWriter
+
+    $XmlWriter.WriteEndElement()
+}
+
+function Write-JUnitTestCaseAttributes($Action, [System.Xml.XmlWriter] $XmlWriter) {
+    $XmlWriter.WriteAttributeString('name', $Action.Name)
+
+    $status = switch ($Action.Result) {
+        Passed {
+            'success'
+        }
+
+        Failed {
+            'failure'
+        }
+
+        default {
+            'skipped'
+        }
+    }
+
+    $XmlWriter.WriteAttributeString('status', $Action.Result)
```

Comment:
Should not this use `$status` instead of `$Action.Result`?

Reply:
I've updated the code a bit to make is more clear what it is used for. 

I'm using the original status for the attribute as its value is not predefined in the schema (keeping terms like "Inconclusive" as part of the junit report). Next, a skipped or failure type adds extra elements which value is predefined in the schema and widgets like the one from GitLab explicitly search for)